BridgeSDK/BridgeAPI/BridgeAPITypes/MachineGeneratedDoNotEdit/_SBBStudyParticipant.m

```diff
@@ -239,6 +239,10 @@ - (void)updateManagedObject:(NSManagedObject *)managedObject withObjectManager:(
         }
     }
 
+    // we do need to fill in the key used for caching this object type, though
```

Comment:
So, types that are cached directly by CacheManager have to have a key path designated as their entity ID key, meaning the key that uniquely identifies an individual object of that type.  This is defined in the entity definition's userInfo field as "entityIDKeyPath". For singletons like StudyParticipant, entityIDKeyPath = "type" so there can only be one object in cache of that type. Well, StudyParticipant is also marked as "encrypted" in the userInfo, so basically its json is encrypted and *that's* what's stored in CoreData, on a "hidden" field called "ciphertext", with the rest left as nil. But then fetch requests to find the object in Core Data by its entity ID would fail for encrypted types (currently only UserProfile and StudyParticipant). So I added a couple of lines to the template to have it store the proper value in the entity ID key for that object, in addition to the ciphertext, so it's still searchable with predicates.